### PR TITLE
Some minor refactors on class `HDFSScanNode`

### PR DESCRIPTION
**Pull request by dirtysalt**
ref: #2661 

This PR tries to:
1. extract `HdfsParquetScanner` to a standalone cpp file
2. create self-owned profile section.
3. move `HdfsScannerCSVReader` from header file to cpp file, since it's only used internally.

### CODE REVIEW

**Comment by dirtysalt** [comment]
There are two children section: Parquet and HdfsIO. I think it's much easier to read and extend profile

image

File: be/src/exec/vectorized/hdfs_scan_node.h [hunk 1]
```diff
@@ -169,24 +172,9 @@ class HdfsScanNode final : public starrocks::ScanNode {
...
-    // reader init
-    RuntimeProfile::Counter* _footer_read_timer = nullptr;
-    RuntimeProfile::Counter* _column_reader_init_timer = nullptr;
-
-    // dict filter
-    RuntimeProfile::Counter* _group_chunk_read_timer = nullptr;
-    RuntimeProfile::Counter* _group_dict_filter_timer = nullptr;
-    RuntimeProfile::Counter* _group_dict_decode_timer = nullptr;
+    HdfsIOProfile _hdfs_io_profile;
+    HdfsParquetProfile _parquet_profile;
```
Comment (DorianZheng) [inline be/src/exec/vectorized/hdfs_scan_node.h:178]: It seems weird to me that `HdfsScanNode` understand something about file format related code. Should we just extract file format related code to other place.

Comment (dirtysalt) [reply be/src/exec/vectorized/hdfs_scan_node.h:178]: I agree. But it's quite hard now. Profile should be attached to some node, and there is only `HDFSSCanNode`.  And theoretically speaking, a single `HDFSScanNode` could read files of many formats like 'parquet' or 'orc'.

This PR is trying to separate format related profile section from general section. But I don't want to make huge change, just for making profile related section better.

I'd like to like your proposal or suggestion.

File: be/src/exec/vectorized/hdfs_scanner_text.h [hunk 2]
```diff
@@ -20,24 +20,7 @@ class HdfsTextScanner final : public HdfsScanner {
     Status _parse_csv(ChunkPtr* chunk);
 
 private:
-    class HdfsScannerCSVReader : public CSVReader {
```
Comment (caneGuy) [inline be/src/exec/vectorized/hdfs_scanner_text.h:23]: These change may also need to be described in the commit message

File: be/src/exec/vectorized/hdfs_scanner.h [hunk 3]
```diff
@@ -185,11 +202,16 @@ class HdfsScanner {
     virtual Status do_get_next(RuntimeState* runtime_state, ChunkPtr* chunk) = 0;
     virtual Status do_init(RuntimeState* runtime_state, const HdfsScannerParams& scanner_params) = 0;
 
+    void in_pending_queue();
```
Comment (caneGuy) [inline be/src/exec/vectorized/hdfs_scanner.h:205]: nit: maybe enter_pending_queue and exit_pending_queue can be easier to understand?

File: be/src/exec/vectorized/hdfs_scanner_text.h (hunk 2 above)
Comment (dirtysalt) [reply be/src/exec/vectorized/hdfs_scanner_text.h:23]: Oh. This is a simple refactor to reduce exposed class in header file. This class is only be used in cpp file.

File: be/src/exec/vectorized/hdfs_scanner.h (hunk 3 above)
Comment (dirtysalt) [reply be/src/exec/vectorized/hdfs_scanner.h:205]: Good idea. Will fix that.

**Review by caneGuy — DISMISSED** [review]
LGTM

**Review by imay — DISMISSED** [review]

File: be/src/exec/vectorized/hdfs_scanner_parquet.cpp [hunk 4]
```diff
@@ -0,0 +1,78 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021 StarRocks Limited.
```
Comment (imay) [inline be/src/exec/vectorized/hdfs_scanner_parquet.cpp:1]: Should rebase the main and change it to 
// This file is licensed under the Elastic License 2.0. Copyright 2021-present StarRocks Limited.

Comment (dirtysalt) [reply be/src/exec/vectorized/hdfs_scanner_parquet.cpp:1]: OK

**Review by imay — DISMISSED** [review]

**Review by caneGuy — DISMISSED** [review]

**Review by DorianZheng — APPROVED** [review]
LGTM

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**